packages/core/src/shared/lsp/lspResolver.ts

```diff
@@ -47,25 +47,40 @@ export class LanguageServerResolver {
 
             const serverResolvers: StageResolver<LspResult>[] = [
                 {
+                    // 1: Use the current local ("cached") LSP server bundle, if any.
                     resolve: async () => await this.getLocalServer(cacheDirectory, latestVersion, targetContents),
                     telemetryMetadata: { id: this.lsName, languageServerLocation: 'cache' },
                 },
                 {
+                    // 2: Download the latest LSP server bundle.
                     resolve: async () => await this.fetchRemoteServer(cacheDirectory, latestVersion, targetContents),
                     telemetryMetadata: { id: this.lsName, languageServerLocation: 'remote' },
                 },
                 {
+                    // 3: If the download fails, try an older, cached version.
                     resolve: async () => await this.getFallbackServer(latestVersion),
                     telemetryMetadata: { id: this.lsName, languageServerLocation: 'fallback' },
                 },
             ]
 
-            return await tryStageResolvers('getServer', serverResolvers, getServerVersion)
+            /**
+             * Example:
+             * ```
+             * LspResult {
+             *   assetDirectory = "<cachedir>/aws/toolkits/language-servers/AmazonQ/3.3.0"
+             *   location = 'cache'
+             *   version = '3.3.0'
+             * }
+             * ```
+             */
+            const resolved = await tryStageResolvers('getServer', serverResolvers, getServerVersion)
```

Comment:
Example values convey a lot of information at a very low cost. Especially important when there is so much indirection and polymorphism.